diff --git a/field_collection.module b/field_collection.module
index 3ab2a76..90c3651 100644
--- a/field_collection.module
+++ b/field_collection.module
@@ -1748,15 +1748,9 @@ function field_collection_field_property_get($entity, array $options, $name, $en
   $values = array();
   if (isset($entity->{$name}[$langcode])) {
     foreach ($entity->{$name}[$langcode] as $delta => $data) {
-      // Wrappers do not support multiple entity references being revisions or
-      // not yet saved entities. In the case of a single reference we can return
-      // the entity object though.
-      if ($field['cardinality'] == 1) {
+      if ($field['cardinality'] == 1 || isset($data['value'])) {
         $values[$delta] = field_collection_field_get_entity($data);
       }
-      elseif (isset($data['value'])) {
-        $values[$delta] = $data['value'];
-      }
     }
   }
   // For an empty single-valued field, we have to return NULL.
